services/movie.py

```diff
@@ -1,40 +1,36 @@
 from django.db.models import QuerySet
-
+from django.db import transaction
 from db.models import Movie
 
 
 def get_movies(
     genres_ids: list[int] = None,
     actors_ids: list[int] = None,
+    title: str = None,
 ) -> QuerySet:
     queryset = Movie.objects.all()
 
     if genres_ids:
         queryset = queryset.filter(genres__id__in=genres_ids)
-
     if actors_ids:
         queryset = queryset.filter(actors__id__in=actors_ids)
+    if title:
+        queryset = queryset.filter(title__icontains=title)
 
     return queryset
 
 
-def get_movie_by_id(movie_id: int) -> Movie:
-    return Movie.objects.get(id=movie_id)
-
-
+@transaction.atomic
 def create_movie(
     movie_title: str,
     movie_description: str,
-    genres_ids: list = None,
-    actors_ids: list = None,
+    genres_ids: list[int] = None,
+    actors_ids: list[int] = None,
```

Comment:
Using mutable default arguments like `list[int]` can lead to unexpected behavior. Consider using `None` as the default value and initializing the list inside the function if needed.